system/jlib/jmetrics.hpp

```diff
@@ -262,4 +273,50 @@ std::shared_ptr<CustomMetric<T>> createCustomMetricAndAddToReporter(const char *
     return pMetric;
 }
 
+
+class jlib_decl ScopedGaugeUpdater
+{
+public:
+    explicit ScopedGaugeUpdater(GaugeMetric &_pGauge, int64_t _amount=1) :
+        gauge{_pGauge},
+        amount{_amount}
+        {
+            gauge.inc(amount);
+        }
```

Comment:
super trivial/style: level of indentation looks a little unusual, i.e. this is at a higher depth than other implementations.
We tend to put the member initializers either on the same line at the ctor (if there's a few), or grouped on next lines, but the inline method impl. at same level as the ctor, e.g.:
```
explicit ScopedGaugeUpdater(GaugeMetric &_pGauge, int64_t _amount=1)
    : gauge{_pGauge}, amount{_amount}
{
    gauge.inc(amount);
} 
```